index.js

```diff
@@ -54,15 +54,21 @@ const oauthPlugin = fp(function (fastify, options, next) {
   const checkStateFunction = options.checkStateFunction || defaultCheckStateFunction
   const startRedirectPath = options.startRedirectPath
 
-  function startRedirectHandler (request, reply) {
-    const state = generateStateFunction(request)
+  function generateAuthorizationUri (requestObject) {
+    const state = generateStateFunction(requestObject)
     const urlOptions = Object.assign({}, callbackUriParams, {
       redirect_uri: callbackUri,
       scope: scope,
       state: state
     })
 
-    const authorizationUri = this[name].oauth2.authorizationCode.authorizeURL(urlOptions)
+    const authorizationUri = oauth2.authorizationCode.authorizeURL(urlOptions)
```

Comment:
why was `this[name]` removed?

Reply:
`this` context seems to be overriden. Its still possible to reference it using `fastify[name]`. I'm not sure if its dependent on `this`. I feel either of the approaches is fine. Anyways let me know what would be the best approach here. I'm personally not 100% sure.

Reply:
After applying my suggestion, this line can be reverted:

tartRedirectHandler is the request handler, so fastiy bind this to the server instance. Changin this flow, here the context is null.

Your change would works in any case, but less var used from upper scope and the code will be more maintainable IMHO

Reply:
`startRedirectHandler` is dependent on having a set `startRedirectPath`. So the handler is never called when using `getAuthorizationUri` only.

Reply:
@Eomm here.

Reply:
> startRedirectHandler is dependent on having a set startRedirectPath

Yes

> So the handler is never called when using getAuthorizationUri only.

This is not clear to me: `getAuthorizationUri` is called by `startRedirectHandler` only
and `startRedirectHandler` is used only as a GET handler.

I would like only to get the point 😅